### ISSUE TITLE
Cannot initiate a private message channel from the bot

### ISSUE THREAD

**Issue by cdelmas** [loafoe/hubot-matteruser#32]
Using this code:
```
robot.hear /plop/i, (res) ->
    robot.messageRoom res.message.user.name, "Hello!"
    robot.send {room: res.message.user.name}, "Hello!"
```
I expect the bot to write a private message to the user. It does work, but only if the user has already open a private message channel to the bot. The bot cannot open it on its initiative the very first time.

I get the following trace:
```
ERROR TypeError: Cannot read property 'toLowerCase' of undefined
  at Brain.userForName (xxxx\basic\node_modules\hubot\src\brain.coffee:120:17, <js>:106:23)
  at Matteruser.send (xxxx\basic\node_modules\hubot-matteruser\src\matteruser.coffee:101:16, <js>:160:31)
  at runAdapterSend (xxxx\basic\node_modules\hubot\src\response.coffee:82:34, <js>:87:50)
  at allDone (xxxx\basic\node_modules\hubot\src\middleware.coffee:44:37, <js>:34:16)
  at xxxx\basic\node_modules\async\lib\async.js:274:13
  at Object.async.eachSeries (xxxx\basic\node_modules\async\lib\async.js:142:20)
  at Object.async.reduce (xxxx\basic\node_modules\async\lib\async.js:268:15)
  at xxxx\basic\node_modules\hubot\src\middleware.coffee:49:7, <js>:37:22
  at _combinedTickCallback (internal/process/next_tick.js:67:7)
  at process._tickCallback (internal/process/next_tick.js:98:9)
```

Reproduced on node 6.9.4, npm 3.10.10.

**Comment by Judokus** [comment]
I guess you are using Mattermost 3.5 or 3.7?

From the readme:

> For example, if you're using Mattermost server version 3.6.0 or 3.6.1, the major version is "3.6", and you need version 3.6 of hubot-matteruser and mattermost-client. Neither version 3.5 or 3.7 will work, since the Mattermost server doesn't yet support API deprecation. See [releases archive](https://github.com/loafoe/hubot-matteruser/releases) for older versions.

**Comment by Judokus** [comment]
Upon further inspection and having the same issue with 3.6... [mattermost-client](https://github.com/loafoe/mattermost-client/blob/master/src/client.coffee#L268) makes a call to `/teams/{teamId}/channels//create_direct` ...

**Comment by loafoe** [comment]
@cdelmas should be fixed in `v3.7.0`, thanks to @Judokus
